drivers/soundwire/bus.c

```diff
@@ -16,27 +16,47 @@
  * Initializes the bus instance, read properties and create child
  * devices.
  */
-int sdw_add_bus_master(struct sdw_bus *bus)
+struct sdw_master_device *
+sdw_add_bus_master(struct sdw_bus *bus, struct device *parent,
+		struct fwnode_handle *fwnode, struct sdw_link_ops *link_ops,
+		void *pdata)
 {
 	struct sdw_master_prop *prop = NULL;
+	struct sdw_master_device *md;
 	int ret;
 
+	md = sdw_master_device_add(parent,
+                                   fwnode,
+                                   link_ops,
+                                   bus->link_id,
+                                   pdata);
+	if (IS_ERR(md)) {
+		pr_err("Could not create link %d\n", bus->link_id);
+	}
+	md->bus = bus;
+	bus->dev = &md->dev;
+
+	if (!bus) {
+		pr_err("SoundWire bus is null\n");
+		return ERR_PTR(-ENODEV);
+	}
```

Comment:
`bus` is a function input parameter. Do you really want to check it *after* you've called `sdw_master_device_add()`? Let alone after dereferencing it multiple times.